src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -176,6 +176,15 @@ CreateInsertSelectPlan(Query *originalQuery,
 					   PlannerRestrictionContext *plannerRestrictionContext)
 {
 	DistributedPlan *distributedPlan = NULL;
+	DeferredErrorMessage *deferredError = NULL;
+
+	deferredError = ErrorIfOnConflictNotSupported(originalQuery);
+	if (deferredError != NULL)
+	{
+		distributedPlan = CitusMakeNode(DistributedPlan);
+		distributedPlan->planningError = deferredError;
+		return distributedPlan;
```

Comment:
It may be worth throwing a hard error here since no amount of recursive planning is going to solve this issue right now.